tests/ClangSharp.PInvokeGenerator.UnitTests/XmlCompatibleUnix/CXXMethodDeclarationTest.cs

```diff
@@ -0,0 +1,933 @@
+    </struct>
+  </namespace>
+</bindings>
+";
+
+            return ValidateGeneratedXmlCompatibleUnixBindingsAsync(inputContents, expectedOutputContents);
+        }
+
+        public override Task MemberCallTest()
+        {
+            var inputContents = @"struct MyStruct
+{
+    int value;
+
+    int MyFunction1()
+    {
+        return value;
+    }
+
+    int MyFunction2()
+    {
+        return MyFunction1();
+    }
+
+    int MyFunction3()
+    {
+        return this->MyFunction1();
+    }
+};
+
+int MyFunctionA(MyStruct x)
+{
+    return x.MyFunction1();
+}
+
+int MyFunctionB(MyStruct* x)
+{
+    return x->MyFunction2();
+}
+";
+
+            var expectedOutputContents = @"<?xml version=""1.0"" encoding=""UTF-8"" standalone=""yes"" ?>
+<bindings>
+  <namespace name=""ClangSharp.Test"">
+    <struct name=""MyStruct"" access=""public"">
+      <field name=""value"" access=""public"">
+        <type>int</type>
+      </field>
+      <function name=""MyFunction1"" access=""public"">
+        <type>int</type>
+        <code>return value;</code>
+      </function>
+      <function name=""MyFunction2"" access=""public"">
+        <type>int</type>
+        <code>return MyFunction1();</code>
+      </function>
+      <function name=""MyFunction3"" access=""public"">
+        <type>int</type>
+        <code>return this.MyFunction1();</code>
+      </function>
+    </struct>
+    <class name=""Methods"" access=""public"" static=""true"">
+      <function name=""MyFunctionA"" access=""public"" static=""true"">
+        <type>int</type>
+        <param name=""x"">
+          <type>MyStruct</type>
+        </param>
+        <code>return x.MyFunction1();</code>
+      </function>
+      <function name=""MyFunctionB"" access=""public"" static=""true"" unsafe=""true"">
+        <type>int</type>
+        <param name=""x"">
+          <type>MyStruct*</type>
+        </param>
+        <code>return x-&gt;MyFunction2();</code>
+      </function>
+    </class>
+  </namespace>
+</bindings>
+";
+
+            return ValidateGeneratedXmlCompatibleUnixBindingsAsync(inputContents, expectedOutputContents);
+        }
+
+        public override Task MemberTest()
+        {
+            var inputContents = @"struct MyStruct
+{
+    int value;
+
+    int MyFunction()
+    {
+        return value;
+    }
+};
+";
+
+            var expectedOutputContents = @"<?xml version=""1.0"" encoding=""UTF-8"" standalone=""yes"" ?>
+<bindings>
+  <namespace name=""ClangSharp.Test"">
+    <struct name=""MyStruct"" access=""public"">
+      <field name=""value"" access=""public"">
+        <type>int</type>
+      </field>
+      <function name=""MyFunction"" access=""public"">
+        <type>int</type>
+        <code>return value;</code>
+      </function>
+    </struct>
+  </namespace>
+</bindings>
+";
+
+            return ValidateGeneratedXmlCompatibleUnixBindingsAsync(inputContents, expectedOutputContents);
+        }
+
+        public override Task NewKeywordTest()
```

Comment:
I suppose this test is a bit redundant given we expect the downstream consumer to determine whether a new keyword is needed.